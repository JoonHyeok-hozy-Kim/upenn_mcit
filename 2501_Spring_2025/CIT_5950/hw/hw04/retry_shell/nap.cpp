#include <unistd.h>  // for sleep
#include <cstdlib>   // for EXIT_SUCCESS
#include <iostream>  // for cout, cerr, and endl

using namespace std;

int main(int argc, char* argv[]) {
  cout << "Sleeping for 10 seconds..." << endl;
  sleep(10);
  cout << "Woke up from 10 second sleep!" << endl;
  return EXIT_SUCCESS;
}
