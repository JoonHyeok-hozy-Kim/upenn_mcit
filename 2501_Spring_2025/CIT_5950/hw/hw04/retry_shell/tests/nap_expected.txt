$ testing to make sure that the shell only reprompts after previous command has finished
$ Sleeping for 10 seconds...
Woke up from 10 second sleep!
$ should have reprompted after nap finished
$ 
